File: backend/src/main/java/com/equisoft/dca/backend/dataobject/model/DataObjectStatus.java
package com.equisoft.dca.backend.dataobject.model;

import com.equisoft.dca.backend.common.model.BaseEnum;

import lombok.Getter;

@Getter
public enum DataObjectStatus implements BaseEnum<Integer> {

	SUCCESS(1, "Success"),
	WARNING(2, "Warning"),
	ERROR(3, "Error");

	private final int id;

	private final String name;

	DataObjectStatus(int id, String name) {
		this.id = id;
		this.name = name;
	}

	@Override
	public Integer getId() {
		return id;
	}
}

File: react/src/lib/domain/entities/DataObjectStatus.ts
import BaseEnum from './BaseEnum';

export default class DataObjectStatus extends BaseEnum {
    public static SUCCESS = new DataObjectStatus('1', 'dataObject:dataObjectStatusEnum:success');
    public static WARNING = new DataObjectStatus('2', 'dataObject:dataObjectStatusEnum:warning');
    public static ERROR = new DataObjectStatus('3', 'dataObject:dataObjectStatusEnum:error');

    public constructor(id?: string, name?: string) {
        super(id, name);
    }
}